### ISSUE TITLE
Add C# Strings

### ISSUE THREAD

**Issue by blockingHD** [codethesaurus/codethesaur.us#365]
# Description
Strings for C# aren't presently on Code Thesaurus. This issue is to add that concept to the system.

# Requirements
Copy web/thesauruses/_meta/strings.json to the right C# directory in web/thesauruses/
Edit the strings.json file to accurately reflect the latest major version of C#
Delete the "name": fields before your PR
Add in "comment": if you wish to explain something in the code
Add in "not-implemented": "true" if the concept doesn't exist in the language. You can then leave "code": blank.

**Comment by blockingHD** [comment]
Adding this so I can take it 😅

**Comment by geekygirlsarah** [comment]
I appreciate it! I haven't gotten to adding issues for strings yet. So go ahead and take it!

Let me know if you have any questions (Though you've done this so you probably know the drill)

**Comment by blockingHD** [comment]
@geekygirlsarah Started this and found that there are somethings that are possible but via other methods:
is_all_alphanumeric - Can be done using Regex
is_all_alphabetical - Can be done using Regex but also can be done at a char level so could loop over the string as well
is_all_uppercase, is_all_lowercase - Can be done at a char level ^^
remove_whitespace - Can't remove all whitespace but `string.Trim()` will remove leading and trailing whitespace.
replace_substring - For a single replacement `Regex.IsMatch(pattern, numberToMatch);` can be used
format_as_integer - What is actually meant by this?
format_number_in_binary, format_number_in_octal - Can be done using `Convert.ToString(int num, int base);` but would need to convert to and integer first and there is no formatter for it. 

Further to this due to how C# manages strings clear_string doesn't actually make much sense as `s1 = string.Empty;` will actually just create a new string so it doesn't clear the string but instead changes the pointer reference. Unless this means how to create and empty string?

**Comment by blockingHD** [comment]
To add just a little bit more fuel to the fire string interning adds a whole lot of nuance to all things stringy. For assign_new_string `var s1 = "hello"; var s2 = s1;` assigns s2 the reference to s1 but that isn't a 'new' instance of the string. This doesn't actually matter as strings are immutable but it depends on how we are defining 'assign'. 

Might have found a rabbit hole and jumped down it...

**Comment by geekygirlsarah** [comment]
Oh this _IS_ a rabbit hole, lol.  But I appreciate these, it helps keep me on my toes but also think through the best way to build the meta files!

For this, I think I was aiming to have string functions (either native or built in to some importable string class/package/library) that would do the things to strings. You can do an absurd amount of things with regexes, but how to solve problems/find patterns with regexes is kind of something I'm leaving up as an exercise for the end user. (Though this does make me think that the strings meta file probably needs a section on regexes on how to call those functions.)

tl;dr document functions, not regex patterns. It'll keep the thesaurus working more thesaurus-ey.
